### DISCUSSION TITLE
App-Only

### DISCUSSION THREAD

**Discussion by ayblanchet** [pnp/pnpframework#163] [Q&A] [answered]
Hi,

I would like create a team with any channels with pnp core sdk + pnp framework.
I can create team channels with pnp core sdk but when I try to create a team with pnp framework I have this error:

System.Private.CoreLib: Exception while executing function: Assistant. PnP.Framework: App-Only is currently not supported, unless you provide a Microsoft Graph Access Token.

I used the same certificate as pnp core sdk with this method:
var authManager = new AuthenticationManager(_settings.ClientId, StoreName.My, StoreLocation.CurrentUser, _settings.CertificateThumbprint, _settings.TenantId);

I try to create team site like this:
TeamSiteCollectionCreationInformation modernteamSiteInfo = new TeamSiteCollectionCreationInformation
{
    Description = "Test modern teamsite description",
    DisplayName = "Test Modern Team Site",
    Alias = "TestModernTeamSite",
    IsPublic = true
    };
var createModernSite = await context.CreateSiteAsync(modernteamSiteInfo); => throw app-only exception

And if I try to create with SiteCreationProperties object I cannot apply GROUP#0 template. 

Can you say me what is wrong? 

Thanks for your help,

**Comment by jansenbe** [comment, marked as answer]
Hi @ayblanchet ,

Does the feedback in #138 help here?

**Reply by ayblanchet** [reply]
Hi, 

To solve my issue, I used :
**UnifiedGroupsUtility.CreateUnifiedGroup** with graph token collect by **AuthenticationManager.GetAccessTokenAsync**.
Is there a way to know what kind of token must be used for a specific method? (ex: CreateUnifiedGroup => Graph token, other method => app only, other method => not app only) 

Best regards,

**Reply by jansenbe** [reply]
It indeed sometimes is not obvious to understand if a Graph call is made under the covers or a SharePoint call
